src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/update/AddToSetOperation.java

```diff
@@ -0,0 +1,158 @@
+    while (it.hasNext()) {
+      Map.Entry<String, JsonNode> entry = it.next();
+      final String modifier = entry.getKey();
+      final JsonNode arg = entry.getValue();
+
+      switch (modifier) {
+        case "$each":
+          eachArg = arg;
+          if (!eachArg.isArray()) {
+            throw new JsonApiException(
+                ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM,
+                ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM.getMessage()
+                    + ": $addToSet modifier $each requires ARRAY argument, found: "
+                    + eachArg.getNodeType());
+          }
+          break;
+
+        default:
+          throw new JsonApiException(
+              ErrorCode.UNSUPPORTED_UPDATE_OPERATION_MODIFIER,
+              ErrorCode.UNSUPPORTED_UPDATE_OPERATION_MODIFIER.getMessage()
+                  + ": $addToSet only supports $each modifier; trying to use '"
+                  + modifier
+                  + "'");
+      }
+    }
+    // For now should not be possible to occur but once we add other modifiers could:
+    if (eachArg == null) {
+      throw new JsonApiException(
+          ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM,
+          ErrorCode.UNSUPPORTED_UPDATE_OPERATION_PARAM.getMessage()
+              + ": $addToSet modifiers can only be used with $each modifier; none included");
+    }
+
+    return new AddToSetAction(propName, eachArg, true);
+  }
+
+  private static boolean hasModifier(ObjectNode node) {
+    Iterator<String> it = node.fieldNames();
+    while (it.hasNext()) {
+      if (looksLikeModifier(it.next())) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean updateDocument(ObjectNode doc, UpdateTargetLocator targetLocator) {
+    boolean modified = false;
+    for (AddToSetAction action : actions) {
+      final String path = action.path;
+      final JsonNode toAdd = action.value;
+
+      UpdateTarget target = targetLocator.findOrCreate(doc, path);
+      JsonNode node = target.valueNode();
+
+      ArrayNode array;
+
+      if (node == null) { // No such property? Add new array
+        array = doc.arrayNode();
+        target.replaceValue(array);
+      } else if (node.isArray()) { // Already array? Append
+        array = (ArrayNode) node;
+      } else { // Something else? fail
+        throw new JsonApiException(
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_TARGET,
+            ErrorCode.UNSUPPORTED_UPDATE_OPERATION_TARGET.getMessage()
+                + ": $addToSet requires target to be ARRAY; value at '"
+                + path
+                + "' of type "
+                + node.getNodeType());
+      }
+
+      if (action.each) {
+        for (JsonNode element : toAdd) {
+          modified |= addToSet(array, element);
+        }
+      } else {
+        modified |= addToSet(array, toAdd);
+      }
+    }
+
+    // Every valid update operation modifies document so need just one:
+    return modified;
```

Comment:
missing int test with modified zero

Reply:
Ok yes good idea; although unit test covers part of it, need to verify everything is linked. Will add.